### PR TITLE
AG-110 - Support custom transaction isolation levels

### PR DESCRIPTION
**Pull request by barreiro**

### CODE REVIEW

**Comment by barreiro** [comment]
@jesperpedersen I though about that, but it breaks backwards compatibility.

**Comment by jesperpedersen** [comment]
Sure, go ahead and merge as is

File: agroal-api/src/main/java/io/agroal/api/configuration/AgroalConnectionFactoryConfiguration.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     Class<?> connectionProviderClass();
 
-    TransactionIsolation jdbcTransactionIsolation();
+    IsolationLevel jdbcTransactionIsolation();
```
Comment (dmlloyd) [inline agroal-api/src/main/java/io/agroal/api/configuration/AgroalConnectionFactoryConfiguration.java]: This breaks binary compatibility, if that matters.

File: agroal-api/src/main/java/io/agroal/api/configuration/supplier/AgroalConnectionFactoryConfigurationSupplier.java [hunk 2]
```diff
@@ -98,12 +99,18 @@ public AgroalConnectionFactoryConfigurationSupplier connectionProviderClassName(
         }
     }
 
-    public AgroalConnectionFactoryConfigurationSupplier jdbcTransactionIsolation(TransactionIsolation transactionIsolationLevel) {
+    public AgroalConnectionFactoryConfigurationSupplier jdbcTransactionIsolation(IsolationLevel transactionIsolationLevel) {
```
Comment (dmlloyd) [inline agroal-api/src/main/java/io/agroal/api/configuration/supplier/AgroalConnectionFactoryConfigurationSupplier.java]: This also breaks binary compatibility; you could add an overload though.

File: agroal-api/src/main/java/io/agroal/api/configuration/AgroalConnectionFactoryConfiguration.java (hunk 1 above)
Comment (barreiro) [reply agroal-api/src/main/java/io/agroal/api/configuration/AgroalConnectionFactoryConfiguration.java]: I don't think it matters. This interface is not intended to be called/implemented outside agroal.

File: agroal-api/src/main/java/io/agroal/api/configuration/supplier/AgroalConnectionFactoryConfigurationSupplier.java (hunk 2 above)
Comment (barreiro) [reply agroal-api/src/main/java/io/agroal/api/configuration/supplier/AgroalConnectionFactoryConfigurationSupplier.java]: On a second though, yes, this method does not need to be relaxed.

**Pull request merged**